src/lib.rs

```diff
@@ -53,7 +53,8 @@ let storage = MemStorage::default();
 config.validate().unwrap();
 // We'll use the built-in `MemStorage`, but you will likely want your own.
 // Finally, create our Raft node!
-let mut node = RawNode::new(&config, storage, vec![]).unwrap();
+storage.initialize_with_config(&config);
```

Comment:
Will there be cased where a user wants to create a storage and then call `initialize_with_config` on it? Or will it always be just create and then run `initialize_with_config`? Maybe having `Storage::new_with_config` which constructs the new storage? :thinking: